File: common-core/src/main/java/io/quartic/geojson/MultiLineString.java
package io.quartic.geojson;


import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.quartic.common.SweetStyle;
import org.immutables.value.Value;

import java.util.List;

@SweetStyle
@Value.Immutable
@JsonSerialize(as = MultiLineStringImpl.class)
@JsonDeserialize(as = MultiLineStringImpl.class)
public interface MultiLineString extends Geometry {
    List<List<List<Double>>> coordinates();

    @Override
    default <T> T accept(GeometryVisitor<T> geometryVisitor) {
        return geometryVisitor.visit(this);
    }
}

File: common-core/src/main/kotlin/io/quartic/geojson/MultiLineString.kt
package io.quartic.geojson

data class MultiLineString @JvmOverloads constructor(
        val coordinates: List<List<List<Double>>>,
        // TODO
        val crs: Map<String, Any>? = emptyMap(),
        val bbox: List<Double>? = emptyList()
) : Geometry
